LogicLayer/GameEvents/GameEventService.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Linq;
+using GameBoard.DataLayer.Repositories;
+using GameBoard.LogicLayer.GameEvents.Dtos;
+using GameBoard.DataLayer.Entities;
+using GameBoard.DataLayer.Enums;
+using JetBrains.Annotations;
+using Microsoft.EntityFrameworkCore;
+
+namespace GameBoard.LogicLayer.GameEvents
+{
+    internal class GameEventService : IGameEventService
+    {
+        private readonly IGameBoardRepository _repository;
+
+        public GameEventService(IGameBoardRepository repository)
+        {
+            _repository = repository;
+        }
+
+        public async Task CreateGameEventAsync(CreateGameEventDto requestedGameEvent)
+        {
+            var creator = await _repository.ApplicationUsers.SingleAsync(
+                ApplicationUser.UserNameEquals(requestedGameEvent.CreatorUserName));
+
+            var creatorParticipation = new GameEventParticipation
+            {
+                ParticipantId = creator.Id,
+                ParticipationStatus = ParticipationStatus.Creator
+            };
+            var gameEvent = new GameEvent
+            {
+                Name = requestedGameEvent.Name,
+                MeetingTime = requestedGameEvent.MeetingTime,
+                Place = requestedGameEvent.Place,
+                Games = requestedGameEvent.Games
+                    .Select(g => new Game { Name = g, GameStatus = GameStatus.ExistsOnTheList}).ToList(),
+                Participations = new List<GameEventParticipation>()
+            };
+
+            gameEvent.Participations.Add(creatorParticipation);
+            _repository.GameEvents.Add(gameEvent);
+
+            await _repository.SaveChangesAsync();
+        }
+
+        public async Task EditGameEventAsync(EditGameEventDto editedEvent)
+        {
+            var gameEventAndGames = await _repository.GameEvents
+                .Where(ge => ge.Id == editedEvent.Id)
+                .Select(ge => new
+                {
+                    GameEvent = ge,
+                    Games = ge.Games.Where(g => g.GameStatus == GameStatus.ExistsOnTheList)
+                })
+                .SingleAsync();
```

Comment:
These are two separate queries anyway, so there's no point in doing it in one LINQ statement and hiding it in a `gameEventAndGames`. Just do two LINQs assigning to `gameEvent` and `games`.

Reply:
Did it.

Reply:
The first of two saves seems redundant.

Also you can split this method into at least two, one would handle all Games related stuff.

Reply:
I split it into 3 functions. However I cannot figure out how to remove the first save. If you tried to remove the first save, then added games "1", "2", "3" when creating an event, then added game "3" between "1" and "2" when editing that event, an exception would be thrown.

Reply:
Acctually adding games "1", "2", then swaping them also throws an exception.

Reply:
Could you now check if it's ok.

Reply:
Looks better.